src/AccessControl.js

```diff
@@ -0,0 +1,187 @@
+import { ethers } from 'ethers';
+import { useCallback, useEffect, useState } from 'react';
+import { useSynthetix } from './useSynthetix';
+
+function AccessControl() {
+  const [synthetix, updateSynthetix] = useSynthetix();
+  const { contract, permissions, walletAddress } = synthetix;
```

Comment:
For the future ease of search I can recommend not exploding the state but use it exclusively as `synthetix.contract` and so on. This makes it very easy to search, replace, rename when needed and lets you skip potential name collisions with local variables.

Reply:
fixed